src/Compilers/Core/MSBuildTask/LocateEditorConfigFiles.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public sealed class LocateEditorConfigFiles : Task
+    {
+        [Required]
+        public ITaskItem[] InputFiles { get; set; }
+
+        [Output]
+        public ITaskItem[] EditorConfigFiles { get; private set; }
+
+        public LocateEditorConfigFiles()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        public override bool Execute()
+        {
+            try
+            {
+                ExecuteCore();
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Log.LogErrorFromException(ex);
+                return false;
+            }
+        }
+
+        private void ExecuteCore()
+        {
+            var directoriesAddedToQueue = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            var directoriesToVisit = new Queue<DirectoryInfo>();
+            var editorConfigFiles = new List<ITaskItem>();
+
+            void addNewDirectoryIfNotAlreadyAdded(DirectoryInfo directory)
+            {
+                // If we're being passed files that don't exist, we'll just skip them and let the compiler handle the actual reporting.
+                if (directory.Exists)
+                {
+                    if (directoriesAddedToQueue.Add(directory.FullName))
+                    {
+                        directoriesToVisit.Enqueue(directory);
+                    }
+                }
+            }
+
+            foreach (var inputFile in InputFiles)
+            {
+                addNewDirectoryIfNotAlreadyAdded(new DirectoryInfo(Path.GetDirectoryName(inputFile.ItemSpec)));
+            }
+
+            while (directoriesToVisit.Count > 0)
+            {
+                var directory = directoriesToVisit.Dequeue();
+                var editorConfigFilePath = Path.Combine(directory.FullName, ".editorconfig");
```

Comment:
Need to consider the worst here: that we're on a case sensitive file system. 

Note: worst is not Linux because Linux can be case insensitive.  The worst is a path where different parts have different casing rules.